### PR TITLE
#479 - Implement Work Package Timeline Status

### LINKED ISSUE #479: Projects - Add Work Package Timeline Status Calculation
## Prerequisites

- [x] #478

## Description

In order to assist engineers and project managers interpreting the current and expected progress of a work package, we want to display an interpretation of the progress called Timeline Status.

## User Story

As an engineer or project manager, I want to understand how the team's progress on a work package compares to the expected progress.

## Business Logic

The work package timeline status will depend on the delta of progress and expected progress. Subtract `expectedProgress` from `progress` to get the `delta`.

- If `delta > 25`, then timeline status is `AHEAD`
- If `delta >= 0`, then timeline status is `ON_TRACK`
- If `delta < 0`, then timeline status is `BEHIND`
- If `delta < -25`, then timeline status is `VERY_BEHIND`

## Proposed Solution

This value needs to be returned by multiple back-end endpoints and displayed in one location on the front-end. As such, the reusable items should probably be written in the `utils` package.

- [ ] create a reusable timeline status enum
- [ ] update the work package interface to include the new `timelineStatus` field
- [ ] create a reusable function to translate from delta to timeline status enum
- [ ] use the function to return a value in the work packages get endpoint
- [ ] use the function to return a value in each work package in the projects get endpoint
- [ ] update the work package details component to display timeline status

### PR DESCRIPTION
**Pull request by jonathan-chen10**
## Changes

- Add TimelineStatus enum for work packages
- Create function to calculate timeline status from given progress and expectedProgress
- Add timeline status to work package front-end

## Test Cases

Utils - calculateTimelineStatus:
- Ahead
- OnTrack
- OnTrack: progress = expectedProgress (edge case)
- Behind
- VeryBehind

Front End:
- Timeline Status in the document

## Screenshots

Screen Shot 2022-03-16 at 4 05 49 PM

## Checklist

- [X] All commits are tagged with the ticket number
- [x] No linting errors
- [x] No newline at end of file warnings
- [X] All code follows repository-configured prettier formatting
- [X] No merge conflicts
- [X] All checks passing
- [X] Screenshots of UI changes (if applicable)
- [X] Remove any not-applicable sections
- [X] Assign the PR to yourself
- [X] PR is linked to the ticket
- [X] No `package-lock.json` changes (unless dependencies have changed)
- [X] Request reviewers & ping on Slack

Closes #479

### CODE REVIEW

**Comment by jonathan-chen10** [comment]
Expected Progress and Timeline Status have been implemented for work packages, but not yet for projects. (The words "Expected Progress" and "Timeline Status" appear on the project page but not their values.)

File: src/utils/src/backend-supports/projects-get-all.ts [hunk 1]
```diff
@@ -42,4 +43,22 @@ const calculatePercentExpectedProgress = (start: Date, weeks: number, status: St
   }
 };
 
-export { projectDurationBuilder, calculateEndDate, calculatePercentExpectedProgress };
+const calculateTimelineStatus = (progress: number, expectedProgress: number): TimelineStatus => {
```
Comment (jamescd18) [inline src/utils/src/backend-supports/projects-get-all.ts:46]: Can we get a js doc comment here to better document the function?

**Review by jamescd18 — CHANGES_REQUESTED** [review]
The is absolutely wonderful! Really clean, very well done! Just one documentation comment to add. Great job!

File: src/utils/src/backend-supports/projects-get-all.ts (hunk 1 above)
Comment (jonathan-chen10) [reply src/utils/src/backend-supports/projects-get-all.ts:53]: Documentation added.

**Review by jamescd18 — APPROVED** [review]
LGTM!
